specs/UserActivityHistory/UserActivityHistoryAPI.md

```diff
@@ -0,0 +1,243 @@
+returned; otherwise, a new UserActivity object will be created and returned. You can then call
+the API GetSession to return a UserActivitySession object that tracks how long the user is engaged
+in that activity. This structure allows multiple sessions to be associated with the same activity,
+representing the case where the user completes that activity a bit at a time - e.g., beginning to
+watch a movie, then pausing, then watching more later. These will be treated as the same singular
+user activity that spans multiple sessions.
+
+UserActivityHistory is a new set of APIs that allow you to query the past 28 days of the user's
+activity history, which will enable you to bring back content that the user has previously been
+interacting with.
+
+# API Pages
+
+## UserActivityHistory class
+
+This class provides static methods that enable you to query the user's activity history.
+This activity history is stored in a database managed by a local service, and these APIs
+call out to that service to retrieve data from the database.
+
+Here is an example usage of the class that will enable you to bring back the webpage for a
+Korean recipe that the user had previously interacted with within the last day:
+
+```c#
+UserActivityHistoryQuery query = new();
+query.Keywords = new string[] { "Korean", "recipe" };
+query.LatestStartTime = DateTime.Now.AddDays(-1);
+
+IList<UserActivityHistoryItem> results = await UserActivityHistory.SearchAsync(
+    new UserActivityHistoryQuery[] { query },
+    UserActivityHistoryOrderBy.DwellTime,
+    maxResults: 1);
+
+UserActivityHistoryItem item = results.FirstOrDefault();
+
+if (item != null)
+{
+    // Now we can use item.ActivationUri to bring back the webpage in the state in which the user
+    // was last viewing it.
+}
+```
+
+## UserActivityHistory.Search method
+
+This method synchronously queries the user's activity history and returns a list of items matching
+the criteria specified in the `queries` parameter. The results are ordered in descending order
+according by the `orderBy` parameter: either by the most recent start times, the most recent
+end times, or the longest time spent on the activity.
+
+Each parameter in `queries` is ORed together in the resulting database query, whereas the contents
+of a single object in the `queries` array are ANDed together.  For example, if you provided two
+queries, each of which contained the keywords "tax", one of which had a content type of
+"application/pdf" and the other had a content type of "image/*", the resulting database query
+would be something along these lines:
+
+```sql
+SELECT * FROM UserActivityHistory WHERE
+    (CONTAINS (DisplayText, '"tax"')) AND
+    ((CONTAINS (ContentType, '"application/pdf"')) OR
+     (CONTAINS (ContentType, '"image/*"')));
+```
+
+## UserActivityHistory.SearchAsync method
+
+This method is an asynchronous version of the `Search` method.
```

Comment:
do we even need this? apps can just spin up a thread, call the sync version and wait for that.

Reply:
here and others -- if we don't have a good reason for async versions, consider removing.